### PR TITLE
plugin/none-ls: Use a generated file

### PR DESCRIPTION
**Pull request by traxys**
This removes the last use of IFD from the plugins. It also removes the runtime check on none-ls sources, moving it at the file generation

### CODE REVIEW

**Review by GaetanLepage — APPROVED** [review]
Diff LGTM

**Comment by GaetanLepage** [comment]
Is it me or the CI has not triggered ?

**Comment by traxys** [comment]
It seems like....

File: flake-modules/updates/none-ls.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  );
+in
+assert lib.assertMsg (lib.length undeclaredTool == 0)
+  "Undeclared tools: ${lib.generators.toPretty { } undeclaredTool}";
+assert lib.assertMsg (lib.length uselesslyDeclaredTool == 0)
+  "Tool is not supported upstream: ${lib.generators.toPretty { } uselesslyDeclaredTool}";
+writeText "efmls-configs-sources.nix" (
+  "# WARNING: DO NOT EDIT\n"
+  + "# This file is generated with packages.<system>.none-ls-builtins, which is run automatically by CI\n"
+  + (lib.generators.toPretty { } builtinSourceNames)
```
Comment (MattSturgeon) [inline flake-modules/updates/none-ls.nix:31]: Glad to have these out of the module eval! :grin:

File: plugins/none-ls/packages.nix [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    "swiftformat"
+    "swiftlint"
+    "textidote"
+    "textlint"
+    "twigcs"
+    "vacuum"
+  ];
+
+  # builtin sources that don't require a package
+  noPackage = [
```
Comment (MattSturgeon) [inline plugins/none-ls/packages.nix:45]: Also an improvement over the old nullable package _or_ false system!

File: plugins/none-ls/servers.nix [hunk 3]
```diff
@@ -5,273 +5,48 @@
...
-  hasBuiltinPackage =
-    source:
-    if builtins.hasAttr source builtinPackages then
-      !(builtins.isBool builtinPackages.${source})
-    else
-      true;
-
-  builtinPackage = source: builtinPackages.${source} or null;
+  # Does this builitin require a package ?
+  hasBuiltinPackage = source: lib.hasAttr source packaged || lib.elem source unpackaged;
```
Comment (MattSturgeon) [inline plugins/none-ls/servers.nix:14]: The name is a little confusing. Would `builtinNeedsPackage` or `requiresPackage` (etc) be better?

**Review by MattSturgeon — APPROVED** [review]
LGTM! I left one comment regarding a name that I think came from the old impl, but if you don't care I won't block on a nit.

File: plugins/none-ls/servers.nix (hunk 3 above)
Comment (traxys) [reply plugins/none-ls/servers.nix:14]: Definitively! I tried finding a better name, but didn't have enough inspiration

**Comment by traxys** [comment]
@Mergifyio queue

**Comment by traxys** [comment]
@Mergifyio queue

**Pull request merged**